chanzuckerberg/idseq-web
# [Maps] Add location_v2 to the dimensions endpoints, left pane filters, and main sample table

### Description
- Add locationV2 dropdown filter to the left sidebar (just based on the full name for now).
- Add locationV2 to the main data table.
- Modify sample and project dimensions endpoints to return locationV2 names. Moved code into LocationHelper sample_dimensions/project_dimensions.
- I decided to continue adding everything as locationV2 (instead of not rendering locationV1) because it makes it easier to verify changes and keep track of what has been updated.
- The test I added to location_helper_test.rb can be improved with some new RSpec factories and tests for SamplesHelper.

### Demo
![Jun-05-2019 20-11-56](https://user-images.githubusercontent.com/5652739/59004599-05650f80-87cf-11e9-820f-d8f01c06f61c.gif)
![Jun-05-2019 20-13-39](https://user-images.githubusercontent.com/5652739/59004600-072ed300-87cf-11e9-819a-24ef8b60dd56.gif)

### Test and Reproduce
- Go to data discovery samples, add "Location v2" column, see locations.
- Use the left pane filters on samples and projects and see Location v2 filter as expected.
